### PR TITLE
daemon: Add new '--envoy-log' command line option and change the default path.

### PR DESCRIPTION
**Pull request by jrajahalme**
So far the Envoy logs were written to the state directory
(/var/run/cilium/cilium-envoy.log), the file system of which was
easily filled with Envoy debug logs, especially if something went
wrong.

Add a new cilium agent command line option to specify where Envoy logs
shoulld be written to. By default this will be
"/var/log/cilium-envoy.log", which is good for production runs. For
testing this default may need to be overridden with an explicit
commmand line option.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
Fixes: #2810

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Review by raybejjani — APPROVED** [review]
Excellent :) Thanks for doing this!

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
@jrajahalme Can you please add "Fix: https://github.com/cilium/cilium/issues/2810" to the first PR message?

**Comment by ianvernon** [comment]
Test failure occurred due to a missing import:
```
22:59:14 PASS
22:59:14 OK: 5 passed
22:59:14 coverage: 8.9% of statements
22:59:14 ok  	github.com/cilium/cilium/pkg/endpoint	0.101s
22:59:24 ?   	github.com/cilium/cilium/pkg/endpointmanager	[no test files]
22:59:29 # github.com/cilium/cilium/pkg/envoy
22:59:29 pkg/envoy/envoy_test.go:76: undefined: filepath in filepath.Join
22:59:29 FAIL	github.com/cilium/cilium/pkg/envoy [build failed]
22:59:29 Makefile:41: recipe for target 'tests-ginkgo-real' failed
22:59:29 make: *** [tests-ginkgo-real] Error 1
22:59:29 Makefile:26: recipe for target 'tests-ginkgo' failed
22:59:29 make: *** [tests-ginkgo] Error 2
```

**Comment by ianvernon** [comment]
test-me-please

**Comment by jrajahalme** [comment]
rebased and added the missing import.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
@manalibhutiyani Note that this PR also changes the default path to `/var/log/cilium-envoy.log`.

**Comment by manalibhutiyani** [comment]
Can you wait for this to merge? https://github.com/cilium/cilium/pull/3047/
Let me know what works best!

**Comment by jrajahalme** [comment]
Can rebase after #3047 is merged!

**Review by tgraf — APPROVED** [review]

**Comment by jrajahalme** [comment]
@manalibhutiyani Thomas just merged this, you'll need to rebase #3047. Not that the log directory has changed, you can keep the default file name as `cilium-envoy.log`, and always try to delete `/var/run/cilium/cilium-envoy.log'.

**Pull request merged**
